central/imagev2/datastore/store/postgres/store.go

```diff
@@ -464,6 +464,12 @@ func (s *storeImpl) upsert(ctx context.Context, obj *storage.ImageV2) error {
 	if err != nil {
 		return err
 	}
+
+	// If the scan is not updated, we need to fill the scan stats from the existing image.
+	if !scanUpdated {
```

Comment:
Move this below `if !metadataUpdated && !scanUpdated {`.  No point in filling in the data if we aren't going to write the image to the database.